services/management/mysql.go

```diff
@@ -78,6 +83,30 @@ func (s *MySQLService) Add(ctx context.Context, req *managementpb.AddMySQLReques
 		if err != nil {
 			return err
 		}
+
+		if len(req.DefaultsFile) != 0 {
+			result, err := s.pfd.ParseDefaultsFile(ctx, req.GetPmmAgentId(), req.GetDefaultsFile(), models.MySQLServiceType)
+			if err != nil {
+				return status.Error(codes.FailedPrecondition, fmt.Sprintf("Defaults file error: %s.", err))
+			}
+
+			// set username and password from parsed defaults file by agent
+			if len(result.Username) != 0 {
+				req.Username = result.Username
+			}
+			if len(result.Password) != 0 {
+				req.Password = result.Password
+			}
+
+			if len(result.Host) != 0 {
+				req.Address = result.Host
+			}
+
+			if result.Port > 0 {
+				req.Port = result.Port
+			}
```

Comment:
based on discussion in Jira ticket we should set values from defaults file only if they weren't passed as a separate field in request.

Reply:
Changed.